### DISCUSSION TITLE
[Script request] LocalAI with GPU passtrough

### DISCUSSION THREAD

**Discussion by FrankieBBBB** [community-scripts/ProxmoxVE#727] [Request script]
### Application Name

LocalAI

### Website

https://github.com/mudler/LocalAI

### Description

LocalAI is the free, Open Source OpenAI alternative. LocalAI act as a drop-in replacement REST API that’s compatible with OpenAI (Elevenlabs, Anthropic... ) API specifications for local AI inferencing. It allows you to run LLMs, generate images, audio (and not only) locally or on-prem with consumer grade hardware, supporting multiple model families.

### Due Diligence

- [X] I have searched existing [scripts](https://community-scripts.github.io/Proxmox/scripts) and found no duplicates.
- [X] I have searched existing [discussions](https://github.com/community-scripts/ProxmoxVE/discussions?discussions_q=) and found no duplicate requests.
- [X] This is not a game-related request.

**Comment by BillyOutlast** [comment]
I am working on bringing a more robust LocalAI implementation to Proxmox via my repository: [BillyOutlast/ProxmoxVED (LocalAI branch)](https://github.com/BillyOutlast/ProxmoxVED/tree/LocalAI).

While LocalAI is a flagship tool for local LLM inference, the community currently lacks a "gold-standard" LXC script that effectively handles the diverse hardware backends (CPU, NVIDIA CUDA, and especially AMD ROCm) in a way that aligns with the community-scripts architecture. I have functional logic for these, but I need help "Proxmox-ifying" it to meet this repo's standards for maintainability and user experience.

## Describe the solution you'd like
A standardized LocalAI LXC script that:

Backend Selection: Allows users to choose their backend (CPU, CUDA, or ROCm) during the installation process.

GPU Passthrough Automation: Incorporates the necessary driver hooks or documentation prompts to ensure the LXC can access the host's GPU resources out of the box.

Modern Architecture: Migrates my current installation logic into the build.func, default.vars, and install.sh pattern used here to ensure it's compatible with the community update scripts. Pending changes to /dev/kfd included
https://github.com/community-scripts/ProxmoxVE/issues/12567#event-23272353190


## Additional context

Upstream Project: https://github.com/mudler/LocalAI

My Current Work: https://github.com/BillyOutlast/ProxmoxVED/tree/LocalAI

Note: This request is separate from my LocalAGI request, as LocalAI serves as the core inference engine/API that LocalAGI (and many other tools) depends on.
